src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC_SelectWptPage.js

```diff
@@ -17,21 +17,54 @@ class CJ4_FMC_SelectWptPage {
+        let waypointsFiltered = [];
+
+        for (let j = 0; j < waypoints.length; j++) {
+            //console.log("checking: " + waypoints[j].ident + " " + waypoints[j].icao);
+            if (waypoints[j].ident == ident) {
+                //console.log("match! adding: " + waypoints[j].icao);
+                waypointsFiltered.push(waypoints[j]);
+                //console.log("pushed: " + waypoints[j].icao);
+            }
+        }
+
+        let currPos = new LatLong(SimVar.GetSimVarValue("GPS POSITION LAT", "degree latitude"), SimVar.GetSimVarValue("GPS POSITION LON", "degree longitude"));
+        //console.log("before sort: " + waypoints[0].icao + ", " + waypoints[1].icao + ", " + waypoints[2].icao + ", " + waypoints[3].icao);
+        waypointsFiltered = waypointsFiltered.sort((a, b) => {
+            let aLatLong = new LatLong(a.infos.coordinates.lat, a.infos.coordinates.long);
+            let bLatLong = new LatLong(b.infos.coordinates.lat, b.infos.coordinates.long);
+            let aDistance = new Number(Avionics.Utils.computeDistance(currPos, aLatLong));
+            let bDistance = new Number(Avionics.Utils.computeDistance(currPos, bLatLong));
+            console.log(a.icao + " " + aDistance);
+            console.log(b.icao + " " + bDistance);
+            return aDistance - bDistance;
+        });
+        //console.log("after sort: " + waypoints[0].icao + ", " + waypoints[1].icao + ", " + waypoints[2].icao + ", " + waypoints[3].icao);
+
         for (let i = 0; i < 5; i++) {
-            let w = waypoints[i + 5 * page];
+            let w = waypointsFiltered[i + 5 * page];
             if (w) {
                 let t = "";
+                let freq = "      ";
+                let region = "";
                 if (w.icao[0] === "V") {
-                    t = " VOR";
+                    let vorType = w.infos.type;
+                    t = vorType == 2 || vorType == 3 ? " VOR-DME " : vorType == 4 || vorType == 5 ? " VORTAC  " : " VOR     ";
+                    freq = w.infos.frequencyMHz.toFixed(2);
+                    region = w.infos.region;
                 }
                 else if (w.icao[0] === "N") {
-                    t = " NDB";
+                    t = " NDB     ";
+                    freq = w.infos.frequencyMHz.toFixed(3);
+                    region = w.infos.region;
                 }
                 else if (w.icao[0] === "A") {
-                    t = " AIRPORT";
+                    t = " AIRPORT ";
+                    region = w.infos.region;
                 }
-                rows[2 * i] = [w.ident + t];
-                rows[2 * i + 1] = [w.infos.coordinates.toDegreeString()];
+                rows[2 * i] = [w.ident.padEnd(5, " ") + t + freq + "[d-text]", region + "[d-text]"];
+                rows[2 * i + 1] = ["  " + w.infos.coordinates.toDegreeString() + "[d-text]"];
                 fmc.onLeftInput[i] = () => {
                     fmc.setMsg("WORKING...");
```

Comment:
i wonder if we should get rid of setting working... here.
it should get set by the caller already an i am not sure if we can run into sticky working... problems here

Reply:
removed